haraka.php

```diff
@@ -4,70 +4,133 @@
  * @author Brian Teeman
  * @copyright (C) 2016 - Brian Teeman
  * @license GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
-**/
+ */
 
-defined( '_JEXEC' ) or die( 'Restricted access' );
-class plgSystemHaraka extends JPlugin
+defined('_JEXEC') or die;
 
+/**
+ * Haraka Plugin by Brian Teeman
+ *
+ * @since  1.0.0
+ */
+class plgSystemHaraka extends JPlugin
 {
+	/**
+	 * Application object.
+	 *
+	 * @var    JApplicationCms
+	 * @since  1.0.0
+	 */
+	protected $app;
+
+	/**
+	 * Load plugin language files automatically
+	 *
+	 * @var    boolean
+	 * @since  1.0.0
+	 */
 	protected $autoloadLanguage = true;
+
+	/**
+	 * Redirect variable
+	 *
+	 * @var    boolean
+	 * @since  1.0.0
+	 */
 	private $redirect = false;
 
-	function plgSystemHarakaSoon($config)
+	/**
+	 * Constructor.
+	 *
+	 * @param   object  &$subject  The object to observe.
+	 * @param   array   $config    An optional associative array of configuration settings.
+	 *
+	 * @return  void
+	 *
+	 * @since   1.0.0
+	 */
+	public function __construct(&$subject, $config)
```

Comment:
Constructor is not needed at all:

* `redirect` is already `false`
* Apparently `redirect` is not used at all